plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1047 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, Frank Dornheim <dornheim@posteo.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+    name: community.general.lxd
+    short_description: Returns Ansible inventory from lxd host
+    description:
+        - Get inventory from the lxd.
+        - Uses a YAML configuration file that ends with 'lxd.(yml|yaml)'.
+    version_added: "3.0.0"
+    author: "Frank Dornheim (@conloos)"
+    options:
+        plugin:
+            description: Token that ensures this is a source file for the 'lxd' plugin.
+            required: true
+            choices: [ 'community.general.lxd' ]
+        url:
+            description:
+            - The unix domain socket path or the https URL for the lxd server.
+            - Sockets in filesystem have to start with C(unix:).
+            - Mostly C(unix:/var/lib/lxd/unix.socket) or C(unix:/var/snap/lxd/common/lxd/unix.socket).
+            default: unix:/var/snap/lxd/common/lxd/unix.socket
+            type: str
+        client_key:
+            description:
+            - The client certificate key file path.
+            aliases: [ key_file ]
+            default: $HOME/.config/lxc/client.key
+            type: path
+        client_cert:
```

Comment:
This - and a couple of other parameters - seem to be recurring in the `lxc`/`lxd` landscape. Probably on another PR but I would strongly consider refactoring out those parameters to `module_utils` and `doc_fragments` to make them reusable by the relevant plugins.

Reply:
I want to create a backlog for me. Is creating a feature request and assign them to me the right way?

Reply:
Yes, that's totally fine. Mention that it should be assigned to you in the issue, then I'll do that.

Reply:
On my Backlog:
- review/rewrite lxd module_utils and look on http-handeling (https://github.com/ansible-collections/community.general/issues/2089)
- refactor the inventory and share them between lxd/lxc
- enable caching

I open the Feature Requests in the evening.

Frank